src/Backups/IBackupDataFileNameGenerator.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+namespace DB
+{
+
+struct BackupFileInfo;
+
+/// Ensures consistent file naming in the backup destination during backup creation.
+class IBackupDataFileNameGenerator
+{
+public:
+    virtual ~IBackupDataFileNameGenerator() = default;
+
+    /// Returns the name of this generator (e.g. "first_file_name" or "checksum").
+    virtual std::string getName() const = 0;
+
+    /// Generates a unique and deterministic backup file name
+    /// based on the given file info.
+    virtual std::string generate(const BackupFileInfo & file_info) = 0;
```

Comment:
The concept of an interface class looks a bit over-complicated here. I mean we have only two kinds of generators and these generators are stateless, so we could write something like
```
String generateBackupDataFileName(const BackupFileInfo & file_info, const BackupSettings & backup_settings);
```

Reply:
Will not be possible to pass `BackupSettings` to `BackupImpl`, so, instead of a single abstruction I'll have to carry over parameters for all generators.